File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int n = nums.length;
        for(int i:nums) map.put(i,map.getOrDefault(i,0)+1);
        for(Map.Entry<Integer,Integer> entry:map.entrySet()){
            if(entry.getValue() > n/2) return entry.getKey();
        }
        return -1;
    }
}

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        Dictionary<int,int> map = new Dictionary<int,int>();
        foreach(int num in nums) {
           map[num] = map.GetValueOrDefault(num,0)+1;
        }
        foreach(KeyValuePair<int, int>  pair in map){
            if(pair.Value  > nums.Length/2){
                return pair.Key;
            }
        }
        return -1;
    }
}